### PR TITLE
204 upgrade line crossing demo to savant rs

### LINKED ISSUE #204: Upgrade line-crossing demo to Savant_rs
Use Rust bindings to manage line-crossing. This is for performance reasons.

Example: https://github.com/insight-platform/savant-rs/blob/main/python/primitives/polygon_match.py

The library is available on PyPi: `pip install savant-rs`.

### PR DESCRIPTION
**Pull request by abramov-oleg**

### CODE REVIEW

File: samples/traffic_meter/line_crossing.py [hunk 1]
```diff
@@ -49,9 +49,38 @@ def __init__(self, **kwargs):
...
+            pt_B = Point(*line_cfg['from'][2:])
+            pt_C = Point(*line_cfg['to'][:2])
+            pt_D = Point(*line_cfg['to'][2:])
+            area = PolygonalArea([pt_A, pt_B, pt_D, pt_C], ["from", None, "to", None])
+            if area.is_self_intersecting():
+                # try to correct the polygon by reversing one of the lines
+                area = PolygonalArea(
+                    [pt_A, pt_B, pt_C, pt_D], ["from", None, "to", None]
+                )
+                if area.is_self_intersecting():
```
Comment (bwsw) [inline samples/traffic_meter/line_crossing.py:59]: @abramov-oleg, maybe exit(1) is a proper way to handle the critical error.

Comment (abramov-oleg) [reply samples/traffic_meter/line_crossing.py:59]: Changed to sys.exit(1)

File: samples/traffic_meter/line_crossing.py [hunk 2]
```diff
@@ -79,18 +106,14 @@ def process_frame(self, buffer: Gst.Buffer, frame_meta: NvDsFrameMeta):
                 primary_meta_object = obj_meta
                 break
 
-        if primary_meta_object is not None and frame_meta.source_id in self.line_config:
-
-            line_from = self.line_config[frame_meta.source_id]['from']
-            line_to = self.line_config[frame_meta.source_id]['to']
-
+        if primary_meta_object is not None and frame_meta.source_id in self.areas:
```
Comment (dorgun) [inline samples/traffic_meter/line_crossing.py:109]: @bwsw, I propose creating a task to add a callback for the stream start in NvDsPyFuncPlugin. Currently, we only have one (on_source_eos) at the end of the stream. This way, we wouldn't need to check every time whether the tracker is created. Шnstead, we can simply create it when a new source stream appears.

File: samples/traffic_meter/line_crossing.py [hunk 3]
```diff
@@ -104,41 +127,49 @@ def process_frame(self, buffer: Gst.Buffer, frame_meta: NvDsFrameMeta):
...
-                            '.'.join(
-                                (
-                                    frame_meta.source_id,
-                                    self.target_obj_label,
-                                    direction.name,
-                                )
+                    obj_metas.append(obj_meta)
+
+            track_lines_crossings = lc_tracker.check_tracks(
+                tuple(obj_meta.track_id for obj_meta in obj_metas)
```
Comment (dorgun) [inline samples/traffic_meter/line_crossing.py]: I suggest avoiding unnecessary conversions. Can the method take a list?

**Review by dorgun — CHANGES_REQUESTED** [review]

File: samples/traffic_meter/line_crossing.py (hunk 2 above)
Comment (bwsw) [reply samples/traffic_meter/line_crossing.py:109]: @dorgun we have a task for that functionality: https://github.com/insight-platform/Savant/issues/61

File: samples/traffic_meter/line_crossing.py (hunk 3 above)
Comment (abramov-oleg) [reply samples/traffic_meter/line_crossing.py]: This is a tuple from a generator expression, there's no extra conversions involved, as far as I understand.

Changed the argument to a list comprehension.

**Review by dorgun — APPROVED** [review]

**Review by dorgun — APPROVED** [review]

**Pull request merged**